### DISCUSSION TITLE
MsigPropose has value 0 on message body

### DISCUSSION THREAD

**Discussion by dsbrgg** [filecoin-project/lotus#8899] [General Q&A] [answered]
- **lotus version**
Daemon:  1.16.0-rc1+2k+git.5a67bb0f6+api1.5.0
Local: lotus version 1.16.0-rc1+2k+git.5a67bb0f6

--------------------------------

I've been testing some `msig` operations and I've set up a multisig wallet with threshold of 1 so there would be no need to approve the transactions as well. I've been using `Filecoin.MsigPropose` RPC call and I was thinking it was strange that even though a CID/msg was returning, the value was set to `0`, I've tried to see if I did this operation on the node CLI and the same happened although the balance for the msig actually changes. 

I'm hoping to understand if this is an expected behaviour and if I'm actually looking into the wrong place, here's a CLI step-by-step which hopefully helps trying to understand this:

```
root@6ae071d54b20:/# lotus msig inspect t2noyj6dqwaetbwpbigrov57toph2qmtary2gduda
Balance: 8974 FIL
Spendable: 8974 FIL
Threshold: 1 / 1
Signers:
ID      Address
t01002  t1l25q6wbc36rrcsx2hmdtiag45h7xo2bndzzedba
Transactions:  0

-----------------------

root@6ae071d54b20:/# lotus msig propose --from t1l25q6wbc36rrcsx2hmdtiag45h7xo2bndzzedba t2noyj6dqwaetbwpbigrov57toph2qmtary2gduda t15hpiakowfitnsoe4pyh75hlpsjpykrgyho5b2gy 1000                                                                                                                        
sent proposal in message:  bafy2bzacecatywysmsznvam36bku6gmwwkqvg7jg2rgcf5mvvxqtkziy4efiw
Transaction ID: 3
Transaction was executed during propose
Exit Code: 0
Return Value:

-----------------------

root@6ae071d54b20:/# lotus msig inspect t2noyj6dqwaetbwpbigrov57toph2qmtary2gduda
Balance: 7974 FIL <---- BALANCE CHANGED
Spendable: 7974 FIL
Threshold: 1 / 1
Signers:
ID      Address
t01002  t1l25q6wbc36rrcsx2hmdtiag45h7xo2bndzzedba
Transactions:  0

-----------------------

root@6ae071d54b20:/# lotus chain get-msg bafy2bzacecatywysmsznvam36bku6gmwwkqvg7jg2rgcf5mvvxqtkziy4efiw
{
  "Message": {
    "Version": 0,
    "To": "t2noyj6dqwaetbwpbigrov57toph2qmtary2gduda",
    "From": "t1l25q6wbc36rrcsx2hmdtiag45h7xo2bndzzedba",
    "Nonce": 3,
    "Value": "0",   <------- VALUE IS ZERO ?????
    "GasLimit": 7195676,
    "GasFeeCap": "101724",
    "GasPremium": "100670",
    "Method": 2,
    "Params": "hFUB6d6AKdYqJtk4nH4P/p1vkl+FRNhKADY1ya3F3qAAAABA",
    "CID": {
      "/": "bafy2bzacecx7asems65e3ks2anry37rlcwyxpvtqbhqc6patgaojlavvbz3ts"
    }
  },
  "Signature": {
    "Type": 1,
    "Data": "0PhIQstfT/CNE+rEXrsnoVErwRbR0sL/t676tjOiTK0k6N8blB+YHsWOeMhXFfK1yF9XwP3X1clNQ+XSorG8bwA="
  },
  "CID": {
    "/": "bafy2bzacecatywysmsznvam36bku6gmwwkqvg7jg2rgcf5mvvxqtkziy4efiw"
  }
}
```

**Comment by dsbrgg** [comment, marked as answer]
Found out the information is encoded in the `Params`:
```
root@6ae071d54b20:/# lotus chain decode params t2noyj6dqwaetbwpbigrov57toph2qmtary2gduda 2 hFUB6d6AKdYqJtk4nH4P/p1vkl+FRNhKADY1ya3F3qAAAABA
{
  "To": "t15hpiakowfitnsoe4pyh75hlpsjpykrgyho5b2gy",
  "Value": "1000000000000000000000",
  "Method": 0,
  "Params": null
}
```

I wonder if there's a RPC method which could return the `Message` with the param decoded?

**Reply by LexLuthr** [reply]
https://lotus.filecoin.io/reference/lotus/state/#statedecodeparams might help but you still need to pass the params from the message to this API. Might not be a simple and straightforward solution you are looking for.

**Reply by dsbrgg** [reply]
Appreciate it @LexLuthr ! I found the CLI method and also the RPC as you mentioned. Do you have any idea if there's a way to fetch the Message with the Params already decoded?

**Reply by LexLuthr** [reply]
No, AFAIK. Params are dependent on the actor and method. You can probably write your own code snippet to do it in one step.
